File: typescript_deno/inflow_api.ts
import { baseUrl, headers } from "./constants";

async function getProduct(productId: string) {
    console.log('getting product');
    const url = `${baseUrl}/${productId}`
    const response = await fetch(url, { headers });
    console.log(response);
}

async function createProduct(productId: string) {
    console.log('creating product');
    const product = { productId, itemType: "StockedProduct", trackSerials: false, name: productId };
    const response = await fetch(baseUrl, { headers, method: "put", body: JSON.stringify(product) });
    console.log(response);
}

async function updateProduct(productId: string) {
    console.log('creating product');
    const product = { productId, description: "updated" };
    const response = await fetch(baseUrl, { headers, method: "put", body: JSON.stringify(product) });
    console.log(response);
}

const productId = crypto.randomUUID()
await createProduct(productId)
await getProduct(productId)
await updateProduct(productId)

File: python3/inflow_api.py
import uuid
import requests
import constants
import json

def get_product(productId: uuid.UUID):
    print("getting product")
    print("")
    url = f'{constants.baseUrl}/{str(productId)}'
    response = requests.get(url, headers=constants.headers)
    print(response.json())
    print("")


def create_product(productId: uuid.UUID):
    print("creating product")
    print("")
    product = {"productId": str(productId), "itemType": "stockedProduct",
               "trackSerials": False, "name": str(productId)}
    response = requests.put(constants.baseUrl, data=json.dumps(
        product), headers=constants.headers)
    print(response.json())
    print("")


def update_product(productId: uuid.UUID):
    print("updating product")
    print("")
    product = {"productId": str(productId), "description": "updated"}
    response = requests.get(constants.baseUrl, data=json.dumps(
        product), headers=constants.headers)
    print(response.json())
    print("")

productId = uuid.uuid4()
create_product(productId)
get_product(productId)
update_product(productId)
